File: DCC2024_Artur/January/Day2_Convert_an_Array_Into_a_2D_Array.py
class Solution(object):
    def findMatrix(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        map = dict()
        max = 0
        
        for num in nums:
            if num not in map:
                map[num] = 0
    
            map[num] += 1
            max = map[num] if map[num] > max else max
        
        res = [[] for _ in range(max)]
        
        for val in map:
            for i in range(map[val]):
                res[i].append(val)
                
        return res

File: DCC2024_Artur/January/Day2_Convert_an_Array_Into_a_2D_Array.cpp
#include <iostream>
#include "vector"
#include "unordered_map"

using namespace std;

class Solution
{
public:
    vector<vector<int>> findMatrix(vector<int>& nums)
    {
        unordered_map<int, int> map;
        vector<vector<int>> res;
        res.emplace_back();

        for(int val : nums)
        {
            if(map[val] + 1 > res.size())
                res.push_back({val});
            else
                res[map[val]].push_back(val);

            map[val]++;
        }

        return res;
    }
};


int main()
{
    Solution solution;

    vector<int> test = {1,3,4,1,2,3,1};

    vector<vector<int>> res = solution.findMatrix(test);

    for(auto & re : res)
    {
        for(int j : re)
            cout << j << ' ';
        cout << endl;
    }

}